src/HTTP.jl

```diff
@@ -1,6 +1,9 @@
 module HTTP
 
-export startwrite, startread, closewrite, closeread
+export startwrite, startread, closewrite, closeread, stack, insert, AWS4AuthLayer, 
+    BasicAuthLayer, CanonicalizeLayer, ConnectionPoolLayer, ContentTypeDetectionLayer,
+    DebugLayer, ExceptionLayer, MessageLayer, RedirectLayer, RetryLayer, StreamLayer, 
+    TimeoutLayer
```

Comment:
This is a lot of exports to add.
I think I am down with it though.
With this PR it is going to be much more comfortable building custom stacks of layers,
so makes sense to make the layer types part of the public API

Reply:
We could export them from an artificial `HTTP.Layers` submodule?

Reply:
I think they are fine as is